File: packages/paso-js/src/commands/shared.ts
import { existsSync } from 'fs';
import { parseFile } from '../parser';
import { validate } from '../validator';
import { PasoDeclaration } from '../types';
import { red, yellow, dim } from '../utils/color';

export function loadAndValidate(filePath: string): PasoDeclaration {
  if (!existsSync(filePath)) {
    console.error(red(`File not found: ${filePath}`) + dim(` Run usepaso init to create one.`));
    process.exit(1);
  }
  const decl = parseFile(filePath);
  const results = validate(decl);
  const errors = results.filter((e) => e.level !== 'warning');
  const warnings = results.filter((e) => e.level === 'warning');

  if (errors.length > 0) {
    console.error(red(`Validation failed with ${errors.length} error(s):`));
    for (const err of errors) {
      console.error(`  ${red(err.path)}: ${err.message}`);
    }
    process.exit(1);
  }

  for (const w of warnings) {
    console.error(`  ${yellow('warning')}: ${w.path}: ${w.message}`);
  }

  return decl;
}

export function slugify(name: string): string {
  return (
    name
      .toLowerCase()
      .replace(/[^a-z0-9]+/g, '-')
      .replace(/^-|-$/g, '') || 'usepaso-service'
  );
}

export function mcpConfigSnippet(serviceName: string): string {
  const slug = slugify(serviceName);
  return `
Connect to an MCP client:

  usepaso connect claude-desktop
  usepaso connect cursor
  usepaso connect vscode
  usepaso connect windsurf

Or add "${slug}" manually to your client config. See: usepaso connect --help`;
}

File: packages/paso-py/usepaso/commands/shared.py
import re
import sys

import click

from usepaso.parser import parse_file
from usepaso.validator import validate
from usepaso.utils.color import red, yellow, dim


def load_and_validate(file_path):
    """Load and validate a usepaso.yaml, exit on error."""
    try:
        decl = parse_file(file_path)
    except FileNotFoundError:
        click.echo(red(f"File not found: {file_path}") + dim(" Run usepaso init to create one."), err=True)
        sys.exit(1)
    except Exception as e:
        click.echo(red(f"Failed to parse {file_path}: {e}"), err=True)
        sys.exit(1)

    results = validate(decl)
    errors = [e for e in results if e.level != 'warning']
    warnings = [e for e in results if e.level == 'warning']

    if errors:
        click.echo(red(f"Validation failed with {len(errors)} error(s):"), err=True)
        for error in errors:
            click.echo(f"  {red(error.path)}: {error.message}", err=True)
        sys.exit(1)

    for w in warnings:
        click.echo(f"  {yellow('warning')}: {w.path}: {w.message}", err=True)

    return decl


def slugify(name: str) -> str:
    return re.sub(r'[^a-z0-9]+', '-', name.lower()).strip('-') or 'usepaso-service'


def mcp_config_snippet(service_name):
    slug = slugify(service_name)
    return f"""
Connect to an MCP client:

  usepaso connect claude-desktop
  usepaso connect cursor
  usepaso connect vscode
  usepaso connect windsurf

Or add "{slug}" manually to your client config. See: usepaso connect --help"""
